query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/MigrationTest.java

```diff
@@ -91,6 +89,63 @@ public void testQuery() {
         executionContext);
   }
 
+  @Test
+  public void testQuerySelectionUsingMapAttributeWithSubPath() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createComplexAttributeExpression("Span.tags", "span.kind"));
+    ViewDefinition viewDefinition = getDefaultViewDefinition();
+    defaultMockingForExecutionContext();
+
+    assertPQLQuery(
+        builder.build(),
+        "Select mapValue(tags__KEYS,'span.kind',tags__VALUES) FROM SpanEventView "
+            + "where "
+            + viewDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "'",
+        viewDefinition,
+        executionContext);
+  }
+
+  @Test
+  public void testQuerySelectionUsingMapAttributeWithSubPathWithColumnAttribute() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createComplexAttributeExpression("Span.id", "span.kind"));
+    ViewDefinition viewDefinition = getDefaultViewDefinition();
+    defaultMockingForExecutionContext();
+
+    assertPQLQuery(
+        builder.build(),
+        "Select span_id FROM SpanEventView "
```

Comment:
So I'm assuming the view def doesn't have `span.id` as a map col, making this query invalid. I'd expect this case to error. Whether it forms an invalid query and sends it to pinot, erroring out there or we do it ourselves - not sure. Let's be consistent with what our current behavior is.

Reply:
This is the example where span.id isn't map attribute and so is treated like a normal column.

Reply:
Rather than responding to a different query then asked, it should error. The only question is - do we form the query as requested and let pinot error (my pref) or do the validation ourrselves.

Reply:
I guess this was discussed and it was pointed to treat such a case as normal column.

Reply:
Can you point me to that discussion? Not sure if it was misinterpreted, or there's a longer conversation to be had but if we can't answer the query as asked, we should not just answer some other query.

Reply:
I guess this was discussed offline.

Reply:
This should be an error scenario. we should throw an IllegalArugment exception for normal attribute with subPath. Only complex attribute like map, array can have subPath.

Reply:
resolved